app/test_server.py

```diff
@@ -7,7 +7,63 @@
 def test_small_test():
     small_test()
 """
+import requests
 
 
 def test_true():
     assert True, "We have a problem!"
+
+
+def test_invalid_url_api_call():
+    response = requests.get('http://localhost:7001/api/v1/search/invalid_url')
+    assert response.json()['Status Code'] == 404
+
+
+def test_google_api_call():
```

Comment:
engines = """ask ubaidu bing duckduckgo exalead google quora yahoo yandex
             tyoutube"""


def make_engine_api_call(engine_name):
    url = 'http://localhost:7001/api/v1/search/' + engine_name
    assert requests.get(url).json()['Status Code'] == 400, engine_name


def test_engine_api_calls(engine_names=None):
    for engine_name in (engine_names or engines.lower().split()):
        make_engine_api_call(engine_name)